osu.Game.Rulesets.Osu/Edit/OsuSelectionHandler.cs

```diff
@@ -41,6 +43,54 @@ protected override void OnOperationEnded()
         /// </summary>
         private Vector2? referenceOrigin;
 
+        public override bool HandleReverse()
+        {
+            var hitObjects = selectedMovableObjects;
+
+            double endTime = hitObjects.Max(h => h.GetEndTime());
+            double startTime = hitObjects.Min(h => h.StartTime);
+
+            bool moreThanOneObject = hitObjects.Length > 1;
+
+            foreach (var h in hitObjects)
+            {
+                if (moreThanOneObject)
+                    h.StartTime = endTime - (h.GetEndTime() - startTime);
+
+                if (h is Slider slider)
+                {
+                    var points = slider.Path.ControlPoints.ToArray();
+                    Vector2 endPos = points.Last().Position.Value;
+
+                    slider.Path.ControlPoints.Clear();
+
+                    slider.Position += endPos;
+
+                    PathType? lastType = null;
+
+                    for (var i = 0; i < points.Length; i++)
+                    {
+                        var p = points[i];
+                        p.Position.Value -= endPos;
+
+                        // propagate types forwards to last null type
+                        if (i == points.Length - 1)
+                            p.Type.Value = lastType;
+                        else if (p.Type.Value != null)
+                        {
+                            var newType = p.Type.Value;
+                            p.Type.Value = lastType;
+                            lastType = newType;
+                        }
+
+                        slider.Path.ControlPoints.Insert(0, p);
```

Comment:
As a side note, since this is the second time this has been done - it's much more efficient to do bulk operations with control points, including adding of control points, because the slider path is pretty much recalculated immediately with each change.